examples/cms/src/data.ts

```diff
@@ -0,0 +1,180 @@
+import type { CollectionItemData, ManagedCollection, ManagedCollectionField } from "framer-plugin"
+
+import { framer } from "framer-plugin"
+import { PLUGIN_KEYS } from "./constants"
+
+export type ManagedCollectionFieldType = ManagedCollectionField["type"]
+export type ManagedCollectionFields = readonly ManagedCollectionField[]
+export type FieldIds = ReadonlySet<ManagedCollectionField["id"]>
+
+type DataSourceField =
+    | {
+          type: ManagedCollectionFieldType
+      }
+    | {
+          type: "html"
+      }
+
+export interface DataSource {
+    id: string
+    fields: Record<string, DataSourceField>
+    items: Record<string, unknown>[]
+}
+
+export function getDataSourcesIds(): string[] {
+    return ["articles", "categories"]
+}
+
+export async function getDataSource(dataSourceId: string, abortSignal?: AbortSignal): Promise<DataSource> {
+    return await fetch(`/datasources/${dataSourceId}.json`, { signal: abortSignal }).then(response => response.json())
+}
+
+export function computeFieldsFromDataSource(dataSource: DataSource): ManagedCollectionFields {
+    const fields: ManagedCollectionField[] = []
+
+    for (const [fieldId, field] of Object.entries(dataSource.fields)) {
+        switch (field.type) {
+            case "html":
+                fields.push({
+                    id: fieldId,
+                    name: fieldId,
+                    type: "formattedText",
+                    userEditable: false,
+                })
+                break
+            case "string":
+            case "number":
+            case "boolean":
+            case "color":
+            case "formattedText":
+                fields.push({
+                    id: fieldId,
+                    name: fieldId,
+                    type: field.type,
+                    userEditable: false,
```

Comment:
`userEditable` is optional and something we are probably going to deprecate, shall we just not include it in the example?

Reply:
It is not optional yet:
```ts
/**
 * Any kind of collection field definition that was created by a plugin and is
 * supported by the API.
 */
type ManagedCollectionField = SupportedCollectionField & {
    /** Is the user able to edit the field within the UI. */
    userEditable: boolean;
};
```

Reply:
Oh no. Let's make it optional ASAP

Reply:
Mhhh, actually, I think we should use the type

```ts
export type EditableManagedCollectionField = SupportedCollectionField &
    Partial<Pick<ManagedCollectionField, "userEditable">>
```

Reply:
But it is not exported by `framer-plugin`